projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.navigation.spec.ts

```diff
@@ -1192,6 +1192,52 @@ describe('IgxHierarchicalGrid Smaller Child Navigation #hGrid', () => {
         expect(secondChildCell.focused).toBe(true);
     }));
 });
+describe('IgxHierarchicalGrid Specific Scenarios Navigation', () => {
+    configureTestSuite();
+    let fixture;
+    let hierarchicalGrid: IgxHierarchicalGridComponent;
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                IgxHierarchicalGridSpecificScenarioComponent
+            ],
+            imports: [
+                NoopAnimationsModule, IgxHierarchicalGridModule]
+        }).compileComponents();
+    }));
+    beforeEach(async(() => {
+        fixture = TestBed.createComponent(IgxHierarchicalGridSpecificScenarioComponent);
+        fixture.detectChanges();
+        hierarchicalGrid = fixture.componentInstance.hgrid;
+        setupHierarchicalGridScrollDetection(fixture, hierarchicalGrid);
+    }));
+
+    it('should navigate to the first cell of next row using Tab from last cell in the row above', (async () => {
+        const row = hierarchicalGrid.getRowByIndex(0);
+        (row as IgxHierarchicalRowComponent).toggle();
+        await wait(100);
+        fixture.detectChanges();
+
+        const cell = hierarchicalGrid.getCellByColumn(2, 'ProductName');
+        cell.nativeElement.focus();
+        await wait(100);
+        fixture.detectChanges();
+
+        const keyboardEvent = new KeyboardEvent('keydown', {
+            key: 'Tab'
+        });
+        cell.dispatchEvent(keyboardEvent);
```

Comment:
There is defined function for keyboard iterations:
UIInteractions.triggerKeyDownEvtUponElem('Tab', cell.nativeElement, true);